lib/stm32/f3/flash.c

```diff
@@ -50,14 +50,105 @@
 /*---------------------------------------------------------------------------*/
 /** @brief Clear All Status Flags
 
-Clears program error, end of operation, busy flags.
-*/
+ Program error, end of operation, write protect error, busy.
+ */
 
 void flash_clear_status_flags(void)
 {
 	flash_clear_pgperr_flag();
 	flash_clear_eop_flag();
+	flash_clear_wrprterr_flag();
 	flash_clear_bsy_flag();
 }
+
+/*---------------------------------------------------------------------------*/
+/** @brief Read All Status Flags
+
+ The programming error, end of operation, write protect error and busy flags
+ are returned in the order of appearance in the status register.
+
+ @returns uint32_t. bit 0: busy, bit 2: programming error, bit 4: write protect
+ error, bit 5: end of operation.
+ */
```

Comment:
Probably better to just refer to the bits by name, or with @ref and use grouping doxygen around the flags, rather than trying to explicitly number the bits here.